File: AngularBasicsExam/src/app/entity/Member.ts
import {Booking} from "./Booking";

export class Member {
  id: number;
  name: string;
  preName: string;
  username: string;
  password: string;
  department: string;
  bookings: Booking[];
  urlToProfilePicture: string;


  constructor(id: number, name: string, preName: string, username: string, password: string, department: string, bookings: Booking[], urlToProfilePicture: string) {
    this.id = id;
    this.name = name;
    this.preName = preName;
    this.username = username;
    this.password = password;
    this.department = department;
    this.bookings = bookings;
    this.urlToProfilePicture = urlToProfilePicture;
  }

  addBooking(booking: Booking) {
    this.bookings.push(booking);
  }
  // getBookingsTimeByDate(date:string): number {
  //   return this.bookings.filter(booking => booking.date === date).reduce((a, b) => a + b.hours, 0);
  // }
}

File: AngularexamBackend/src/main/java/AngularexamBackend/AngularexamBackend/Entities/Member.java
package AngularexamBackend.AngularexamBackend.Entities;


import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@Entity
@Table
public class Member {
    @Id
    Long id;
    String name;
    String preName;
    String userName;
    String password;
    String department;
    double workingHours;
    String urlToProfilePicture;
}
